File: javascript/2015/day01/task1-2.js
const helper = require('./helper');
const utils = require('../../utils');
var data = utils.getData()[0];

function handleCharacter(char) {
	if (char == '(') return 1;

	return -1;
}

var floor = 0;
var entered_basement = false;

for (var i = 0; i < data.length; i++) {
	floor += handleCharacter(data[i]);

	if (floor == -1 && entered_basement === false) {
		entered_basement = i+1;
	}

}

console.log('Task 1:', floor);
console.log('Task 2:', entered_basement);

File: python/2015/day01/task1-2.py
def get_data():
    with open("data.txt") as file:
        return file.readline().rstrip()


def tasks(data):
    floor = 0
    entered_basement = False
    commands = {"(": 1, ")": -1}

    for i, character in enumerate(data):
        floor += commands[character]

        if not entered_basement and floor < 0:
            entered_basement = i + 1

    return floor, entered_basement


data = get_data()
floor, entered_basement = tasks(data)

print(f"Task 1: {floor}")
print(f"Task 2: {entered_basement}")
